src/components/AddTask.jsx

```diff
@@ -0,0 +1,71 @@
+import { ReactComponent as DeleteIcon } from 'assets/delete.svg';
+import Button from 'components/ui/Button';
+import { useEffect, useRef, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { addTodo, setIsAddingTask } from 'redux/actions/TodoAction';
+import { KEY_ENTER, RESPONSE_OK } from 'utils/constants';
+import { validate } from 'utils/helpers/index';
+
+function AddTask() {
+  const [error, setError] = useState(null);
+  const [taskDetails, setTaskDetails] = useState('');
+  const dispatch = useDispatch();
+  const textAreaRef = useRef(null);
+
+  function onSubmit(e) {
+    e.preventDefault();
+    const validateDetails = validate(taskDetails);
+
+    if (validateDetails.status === RESPONSE_OK) {
+      setError(null);
+      dispatch(addTodo(validateDetails.text));
+      dispatch(setIsAddingTask(false));
+      setTaskDetails('');
```

Comment:
Could you please re-check this validation code ? As we are not returning if `validateDetails.status === RESPONSE_OK`, so setError might call twice.